code/modules/lighting/lighting_corner.dm

```diff
@@ -39,49 +40,48 @@ GLOBAL_LIST_INIT(LIGHTING_CORNER_DIAGONAL, list(NORTHEAST, SOUTHEAST, SOUTHWEST,
 	// Issue being that the only way I could think of doing it was very messy, slow and honestly overengineered.
 	// So we'll have this hardcode instead.
 	var/turf/T
-	var/i
 
 	// Diagonal one is easy.
 	T = get_step(new_turf, diagonal)
 	if (T) // In case we're on the map's border.
-		if (!T.corners)
-			T.corners = list(null, null, null, null)
-
-		masters[T]   = diagonal
-		i            = GLOB.LIGHTING_CORNER_DIAGONAL.Find(turn(diagonal, 180))
-		T.corners[i] = src
+		save_master(T, diagonal)
 
 	// Now the horizontal one.
 	T = get_step(new_turf, horizontal)
 	if (T) // Ditto.
-		if (!T.corners)
-			T.corners = list(null, null, null, null)
-
-		masters[T]   = ((T.x > x) ? EAST : WEST) | ((T.y > y) ? NORTH : SOUTH) // Get the dir based on coordinates.
-		i            = GLOB.LIGHTING_CORNER_DIAGONAL.Find(turn(masters[T], 180))
-		T.corners[i] = src
+		save_master(T, ((T.x > x) ? EAST : WEST) | ((T.y > y) ? NORTH : SOUTH)) // Get the dir based on coordinates.
```

Comment:
Would it be worth making this a define?

Reply:
My mind will melt if i stare at this code any longer, you can pr it to my branch or pr when you merge this.

Its a target for sure, now that corner datums get created later in the process and self clean up when lights go away.

Reply:
aight, that's a good enough reason for me